api/endpoint.go

```diff
@@ -451,5 +513,11 @@ func (e *Endpoint) AttributesPost(ctx *gin.Context) {
 		ctx.AbortWithError(http.StatusBadRequest, err)
 		return
 	}
+
+	if err := parseSas(&request.Vds, &request.Sas); err != nil {
+		ctx.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
```

Comment:
I meant that nothing prevents us from creating `parsePostRequest` as well, what Erlend does now in a different PR.

But why can't we move this check to the `parseSomeRequest` functions?
I am bad at go, but does something prevent us after parsing the data to cast `interface` to `RequestedResource` and call `parse(extract)Sas` in the function itself?
Conceptually I think it makes sense - we are still parsing the request by extracting the sas part, so it can be part of the `parseSomeRequest` function.

Reply:
I have moved the new function call inside the `parsePostRequest` and `parseGetRequest` functions. 
We cannot cast the object of `interface{}` into `RequestResource` directly as it is a struct of struct and it becomes very complicated. Instead of that, we have created another interface to call the new function with a pointer so that we can modify the struct values.